### PR TITLE
61: refactor hardhat as NX executor

### PR DESCRIPTION
**Pull request by saleel**
## Description

- Refactor Contracts as a NX project
- Add custom executors to build and deploy contracts

## Related Issue

#61 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

## Notes

- The base branch of this PR is `chore/docker`. This PR should only be merged after #81

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
         "test": "nx test",
         "start:all": "nx run-many --target=serve --all",
         "build:all": "nx run-many --target=build --all",
-        "test:all": "nx run-many --target=test --all --coverage",
+        "test:all": "nx run-many --target=test --projects=api,client,dashboard --coverage",
```
Comment (saleel) [inline package.json:11]: Test in `contracts` are failing (even before changes in PR). Temporarily remove `contracts` project from `test:all`

File: apps/contracts/README.md [hunk 2]
```diff
@@ -7,51 +7,50 @@
 
 ---
 
-## Install
+### Install / Compile
 
-Install the dependencies:
+`Contracts` in a NX app, and installation/building can be done by running respective commands in the project root.
```
Comment (cedoor) [inline apps/contracts/README.md:12]: `in` -> `is`. Typo?

File: apps/contracts/README.md [hunk 3]
```diff
@@ -7,51 +7,50 @@
...
-
-### Compile
-
-Compile the smart contracts with Hardhat:
-
-```bash
-npm run compile
+# in project root
+yarn
+npm yarn build
```
Comment (cedoor) [inline apps/contracts/README.md:17]: `npm yarn build` -> `nx run contracts:build` (?)

**Review by cedoor — CHANGES_REQUESTED** [review]

File: apps/contracts/README.md (hunk 2 above)
Comment (saleel) [reply apps/contracts/README.md]: Yes. Fixed, thanks.

File: apps/contracts/README.md (hunk 3 above)
Comment (saleel) [reply apps/contracts/README.md]: Yea, fixed this as well. I mean't to write`yarn build:all` earlier. I have added commands for building separately and all of them together.

**Review by cedoor — APPROVED** [review]

**Pull request merged**